### PR TITLE
Remove usages of FileSafety class

### PR DESCRIPTION
**Pull request by huoyaoyuan**
As https://github.com/ppy/osu-framework/pull/3004#discussion_r347746661 , the entire class is going to be removed. Since game side is using totally different methods with framework side, just copying implementations of the 2 used methods.

- [x] Depends on ppy/osu-framework#3004 after change.

### CODE REVIEW

File: osu.Game/IO/FileStore.cs [hunk 1]
```diff
@@ -130,5 +130,12 @@ public override void Cleanup()
                 }
             }
         }
+
+        /// <summary>
+        /// Moved from osu.Framework.IO.FileSafety.
+        /// </summary>
+        /// <param name="path">The path to standardise.</param>
+        /// <returns>The standardised path using '/' for directory separator.</returns>
+        internal static string PathStandardise(string path) => path.Replace('\\', '/');
```
Comment (huoyaoyuan) [inline osu.Game/IO/FileStore.cs]: A temporary position not unsuitable for the method to live. May be factored out further.

Comment (peppy) [reply osu.Game/IO/FileStore.cs]: Disagree with this placement. How about just making it an extension method and putting in `ExtensionMethods`?

Comment (huoyaoyuan) [reply osu.Game/IO/FileStore.cs]: Should be sure, but then this PR must be merged together with updating framework.

File: osu.Game/Beatmaps/WorkingBeatmap.cs [hunk 2]
```diff
@@ -83,7 +82,11 @@ protected virtual Track GetVirtualTrack()
         /// <returns>The absolute path of the output file.</returns>
         public string Save()
         {
-            var path = FileSafety.GetTempPath(Guid.NewGuid().ToString().Replace("-", string.Empty) + ".json");
+            // copied from osu.Framework.IO.File.FileSafety.GetTempPath
```
Comment (smoogipoo) [inline osu.Game/Beatmaps/WorkingBeatmap.cs]: unnecessary

**Review by smoogipoo — CHANGES_REQUESTED** [review]

**Review by peppy — DISMISSED** [review]

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**